cherrypy/test/test_static.py

```diff
@@ -193,6 +209,19 @@ def test_static(self):
         #   we just check the content
         self.assertMatchesBody('^Dummy stylesheet')
 
+        # Check a filename with utf-8 characters in it
+        ascii_fn = 'has_utf-8_character_.html'
+        url_quote_fn = 'has_utf-8_character_%E2%98%83.html'
+        cd = '''attachment; filename="%s"; filename*=UTF-8\'\'%s'''
+
+        self.getPage('/serve_file_utf8_filename')
+        self.assertStatus('200 OK')
+        self.assertHeader('Content-Disposition', cd % (ascii_fn, url_quote_fn))
+
+        self.getPage('/serve_fileobj_utf8_filename')
+        self.assertStatus('200 OK')
+        self.assertHeader('Content-Disposition', cd % (ascii_fn, url_quote_fn))
```

Comment:
no need to have a multiline literal here, also let's use `str.format()`:
```suggestion
        expected_content_disposition = (
            'attachment; filename="{!s}"; filename*=UTF-8\'\'{!s}'.
            format(ascii_fn, url_quote_fn)
        )

        self.getPage('/serve_file_utf8_filename')
        self.assertStatus('200 OK')
        self.assertHeader('Content-Disposition', expected_content_disposition)

        self.getPage('/serve_fileobj_utf8_filename')
        self.assertStatus('200 OK')
        self.assertHeader('Content-Disposition', expected_content_disposition)
```